temoa/data_io/hybrid_loader.py

```
@@ -629,6 +637,120 @@ def _load_existing_capacity(
+            vintage_exist_data = sorted({(row[2],) for row in rows_to_load})
+            self._load_component_data(data, model.vintage_exist, vintage_exist_data)
+
+            # Collect existing capacity data indices
+            self.viable_existing_techs = {row[1] for row in rows_to_load}
+            self.viable_existing_rt = {(row[0], row[1]) for row in rows_to_load}
+            self.viable_existing_rtv = {(row[0], row[1], row[2]) for row in rows_to_load}
+
+    def _load_retired_existing_capacity(
+        self,
+        data: dict[str, object],
+        raw_data: Sequence[tuple[object, ...]],
```

Ruff (0.15.20)

[warning] Unused method argument: `raw_data`

(ARG002)